src/dodal/common/visit.py

```diff
@@ -150,3 +151,20 @@ def __call__(self, device_name: str | None = None) -> PathInfo:
         return PathInfo(
             directory_path=self._root, filename=self._filename_provider(device_name)
         )
+
+
+DEFAULT_TEMPLATE = "{device_name}-{instrument}-{scan_id}"
+
+
+class StartDocumentBasedPathProvider(PathProvider):
```

Comment:
Should: Docstrings on this class would be good

Reply:
I have added some now.